### ISSUE TITLE
CORE deposit files showing in ElasticSearch results

### ISSUE THREAD

**Issue by mikethicke** [MESH-Research/commons#210]
adonlon:

>A note that the update yesterday introduced some of the issues on dev to prod (this may be known/expected!)--in particular, there are results with CORE deposit file names that don't resolve to the deposit page. For example, searching Shakespeare on MLA Commons has a result (about five down) titled "shakespeare-transformed-copyright-copyleft-and-shakespeare-after-shakespeare.pdf" which links to https://mla.hcommons.org/humcore_deposit/hc17197/hc17198/ which doesn't resolve / redirects to the HC homepage.

These results should not be appearing. Either stop them from being indexed (preferable), or filter them out from from results.

**Comment by adonlon** [comment]
I came across a search that demonstrates the problem well, in case it's useful for testing. Search "abstracts" and filter to CORE deposits. (The 1st page of results are all deposit files.)

**Comment by mikethicke** [comment]
Humcore deposits with parents (ie. attachments) are supposed to be filtered out during indexing. I think they were being included due to an overly-strict check (0 === false). This has been corrected, though it is difficult to test given how indexing works. I also added a check on returning the results in the hc-suggestions widget and the elasticpress-buddypress plugin.

See:
- https://github.com/MESH-Research/elasticpress-buddypress/commit/3d19c45ed4df1b22a6f8567a039e3272b627494e
- https://github.com/MESH-Research/hc-custom/commit/214cb659479cbc306d7e63cc4ab9e164087799d4
- https://github.com/MESH-Research/hc-suggestions/commit/af5ffe998eee4fe8e69b873e407af4d51670cb26

Note: the hc-suggestion widget does not seem to be working correctly with ElasticPress. It should probably be rewritten to use the elasticpress-buddypress REST API rather than its own.
